src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,222 @@
+            if (type.contains("article") || type.contains("Article")) {
+                return StandardEntryType.Article;
+            }
+            if (type.contains("thesis") || type.contains("Thesis")) {
+                return StandardEntryType.Thesis;
+            }
+            if (type.contains("book") || type.contains("Book")) {
+                return StandardEntryType.Book;
+            }
+        }
+        return StandardEntryType.Misc;
+    }
+
+    // Gets all the authors, separated with the word "and"
+    // For some reason the author field sometimes has extra numbers and letters.
+    private String getAuthor(Node itemElement) {
+        StringJoiner stringJoiner = new StringJoiner(" and ");
+        for (int i = 1; i < itemElement.getChildNodes().getLength(); i += 2) {
+            String next = removeNumbers(itemElement.getChildNodes().item(i).getTextContent()).replaceAll("\\s+", " ");
+            next = next.replace("\n", "");
+            if (next.isBlank()) {
+                continue;
+            }
+            stringJoiner.add(next);
+        }
+        return (stringJoiner.toString().substring(0, stringJoiner.length())).trim().replaceAll("\\s+", " ");
+    }
+
+    // Remove numbers from a string and everything after the number, (helps with the author field).
+    private String removeNumbers(String string) {
+        for (int i = 0; i < string.length(); i++) {
+            if (Character.isDigit(string.charAt(i))) {
+                return string.substring(0, i);
+            }
+        }
+        return string;
+    }
+
+    // In the XML file the publishers node often lists multiple publisher e.g.
+    // <publisher origin="HAL CCSD">HAL CCSD</publisher>
+    // <publisher origin="Elsevier">Elsevier</publisher>
+    // Therefore this function simply gets all of them.
+    private String getPublishers(Node itemElement) {
+        if (itemElement == null) {
+            return "";
+        }
+        StringJoiner stringJoiner = new StringJoiner(", ");
+        for (int i = 0; i < itemElement.getChildNodes().getLength(); i++) {
+            if (itemElement.getChildNodes().item(i).getTextContent().isBlank()) {
+                continue;
+            }
+            stringJoiner.add(itemElement.getChildNodes().item(i).getTextContent().trim());
+        }
+        return stringJoiner.toString();
+    }
+
+    private String getJournal(NodeList list) {
+        // If there is no journal, return an empty string.
+        if (list.getLength() == 0) {
+            return "";
+        }
+        String reference = list.item(list.getLength() - 1).getTextContent();
+        for (int i = 0; i < reference.length(); i++) {
+            if (reference.charAt(i) == ',') {
+                return reference.substring(0, i);
+            }
+        }
+        return "";
+    }
+
+    @Override
+    public void doPostCleanup(BibEntry entry) {
+        IdBasedParserFetcher.super.doPostCleanup(entry);
```

Comment:
the interface has a default method, so if you don't to anything extra, you can remove the. method
the same applies to the method below